### PR TITLE
fix(metrics): Add equality comparison on `MetricField`

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR aims at solving a problem in which `MetricField`s that are considered equal with `__hash()__` were not actually considered equal due to the use of the base implementation of `__eq__()` which under the hood uses the `is` operator.

This fix is needed, in order to properly understand whether the orderby fields are a subset of the select fields by considering every element parameter besides the `alias`.

### CODE REVIEW

**Review by ahmedetefy — COMMENTED** [review]
Can we add a test that would fail without this. Preferable the one that @wmak suggested?

**Comment by iambriccardo** [comment]
> Can we add a test that would fail without this. Preferable the one that @wmak suggested?

Yes, I was thinking about it. Will create it now

File: src/sentry/snuba/metrics/query.py [hunk 1]
```diff
@@ -55,6 +55,13 @@ def __str__(self) -> str:
         metric_name = get_public_name_from_mri(self.metric_mri)
         return f"{self.op}({metric_name})" if self.op else metric_name
 
+    def __eq__(self, other: object) -> bool:
+        # The equal method is called after the hash method to verify for equality of objects to insert
+        # into the set. Because by default "__eq__()" does use the "is" operator we want to override it and
+        # model MetricField's equivalence as having the same hash value, in order to reuse the comparison logic defined
+        # in the "__hash__()" method.
+        return bool(self.__hash__() == other.__hash__())
```
Comment (iambriccardo) [inline src/sentry/snuba/metrics/query.py:63]: I have added `bool()` for the type checking in place in the CI.

Comment (andriisoldatenko) [reply src/sentry/snuba/metrics/query.py:63]: can you please add small test to demonstrate this specific problem ... it's hard to understand from `test_validate_order_by_field_in_select_with_different_alias`.

I mean create a test to compare 2 MetricField's

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**